lib/std/times/durations.nim

```diff
@@ -0,0 +1,314 @@
+#
+#
+#            Nim's Runtime Library
+#        (c) Copyright 2021 Nim contributors
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+## The `std/times/durations` module contains routines and types for dealing with durations of time.
+## It is reexported by the `std/times <times.html>`_ modules
+##
+## A `Duration` represents a duration of time stored as seconds and
+## nanoseconds. It is always fully normalized, so
+## `initDuration(hours = 1)` and `initDuration(minutes = 60)` are equivalent.
+##
+## Arithmetic with a `Duration` is very fast, since it only involves basic arithmetic.
+
+import core {.all.} # std/times/core
+
+include "system/inclrtl"
+
+type
+  Duration* = object
+    ## Represents a fixed duration of time, meaning a duration
+    ## that has constant length independent of the context.
+    ##
+    ## To create a new `Duration`, use `initDuration
+    ## <#initDuration,int64,int64,int64,int64,int64,int64,int64,int64>`_.
+    ## Instead of trying to access the private attributes, use
+    ## `inSeconds <#inSeconds,Duration>`_ for converting to seconds and
+    ## `inNanoseconds <#inNanoseconds,Duration>`_ for converting to nanoseconds.
+    seconds: int64
+    nanosecond: NanosecondRange
+
+  DurationParts* = array[FixedTimeUnit, int64] # Array of Duration parts starts
+
+#
+# Helper procs
+#
+
+{.pragma: operator, rtl, noSideEffect, benign.}
+
+proc normalize(seconds, nanoseconds: int64): Duration =
+  ## Normalize a (seconds, nanoseconds) pair and return it as
+  ## a `Duration`. A normalized `Duration` has a
+  ## positive nanosecond part in the range `NanosecondRange`.
+  result.seconds = seconds + convert(Nanoseconds, Seconds, nanoseconds)
+  var nanosecond = nanoseconds mod convert(Seconds, Nanoseconds, 1)
+  if nanosecond < 0:
+    nanosecond += convert(Seconds, Nanoseconds, 1)
+    result.seconds -= 1
+  result.nanosecond = nanosecond.int
+
+#
+# Duration
+#
+
+const DurationZero* = Duration() ## \
```

Comment:
is there any style guide still recommending `## \` ? if so they should be updated, it doesn't seem necessary (simpler = better)
as can be seen with:

`nim doc -r main`
```nim
const a1* = 10 ## \
  ## ok1
  ## ok2

const a2* = 10
  ## ok1
  ## ok2
```

Reply:
My bad, I forgot to change that (`times` used this before, for some reason).

Reply:
separate cleanup PR welcome; better to separate non-controversial PRs from more difficult ones so that the non-controversial one gets merged fast and doesn't complicate the other one